core/src/test/scala/ai/lum/odinson/foundations/TestIndexWriter.scala

```diff
@@ -72,7 +71,58 @@ class TestOdinsonIndexWriter extends OdinsonTest {
     luceneStringField.head.name shouldEqual ("smth")
     // TODO: should we test more stuff
     indexWriter.close()
-    deleteIndex
+  }
+
+  it should "incrementally append to a new index" in {
+    val appendConf = testConfig.withValue("odinson.index.incremental", ConfigValueFactory.fromAnyRef(true))
+    val indexer = getOdinsonIndexWriter(appendConf)
+    var reader : IndexReader = null
+
+    // add one doc...
+    val docOne : OdinsonDocument = getDocument("alien-species")
+    indexer.addDocument(docOne)
+
+    reader = DirectoryReader.open(FSDirectory.open(indexDir.toPath))
+    reader.numDocs shouldBe 2 // number of lucene entries, not documents
+    reader.close()
+
+    val docTwo = getDocument("ninja-turtles")
+    indexer.addDocument(docTwo)
+
+    reader = DirectoryReader.open(FSDirectory.open(indexDir.toPath))
+    reader.numDocs() shouldBe 5 // number of lucene entries, not documents
+
+    reader.close()
+    indexer.close()
+  }
+
+  it should "incrementally append to an existing index" in {
+    val appendConf = testConfig.withValue("odinson.index.incremental", ConfigValueFactory.fromAnyRef(true))
+    var indexer : OdinsonIndexWriter = getOdinsonIndexWriter(appendConf)
+    var reader : IndexReader = null
+
+    // add one doc...
+    val docOne : OdinsonDocument = getDocument("alien-species")
+    indexer.addDocument(docOne)
+
+    reader = DirectoryReader.open(FSDirectory.open(indexDir.toPath))
+
+    reader.numDocs shouldBe 2 // number of lucene entries, not documents
+    reader.close()
+    indexer.close()
+
```

Comment:
@reynoldsm88 to really test this -- can you add a line here that (a) runs a query, you get N results, (b) then you add the new document, (c) and with the same extractor engine, run the same query and you get N + M results?